activesupport/lib/active_support/core_ext/object/blank.rb

```diff
@@ -112,12 +112,9 @@ class String
   #
   # @return [true, false]
   def blank?
-    # In practice, the majority of blank strings are empty. As of this writing
-    # checking for empty? is about 3.5x faster than matching against the regexp
-    # in MRI, so we call the predicate first, and then fallback.
-    #
-    # The penalty for blank strings with whitespace or present ones is marginal.
-    empty? || BLANK_RE === self
+    # Regex check is slow, only check non-empty strings.
+    # A string not blank if it contains a single non-space string.
+    empty? || !(/[[:^space:]]/ === self)
```

Comment:
I tried benching, and it's in the same ballpark, it keeps on ending up as slower for me. I'm not sure why

```
not a non-whitespace char:    21258.3 i/s
!~                       :    18617.7 i/s - same-ish: difference falls within error
```

Which is weird because it is actually fewer instructions

```
irb(main):162:0* puts RubyVM::InstructionSequence.disasm -> { /[[:^space:]]/ !~ str }
== disasm: #<ISeq:block in irb_binding@(irb)>===========================
== catch table
| catch type: redo   st: 0002 ed: 0013 sp: 0000 cont: 0002
| catch type: next   st: 0002 ed: 0013 sp: 0000 cont: 0013
|------------------------------------------------------------------------
0000 trace            256                                             ( 162)
0002 trace            1
0004 putobject        /[[:^space:]]/
0006 putself
0007 opt_send_without_block <callinfo!mid:str, argc:0, FCALL|VCALL|ARGS_SIMPLE>, <callcache>
0010 opt_send_without_block <callinfo!mid:!~, argc:1, ARGS_SIMPLE>, <callcache>
0013 trace            512
0015 leave
=> nil
irb(main):163:0> puts RubyVM::InstructionSequence.disasm -> { !(/[[:^space:]]/ === str) }
== disasm: #<ISeq:block in irb_binding@(irb)>===========================
== catch table
| catch type: redo   st: 0002 ed: 0016 sp: 0000 cont: 0002
| catch type: next   st: 0002 ed: 0016 sp: 0000 cont: 0016
|------------------------------------------------------------------------
0000 trace            256                                             ( 163)
0002 trace            1
0004 putobject        /[[:^space:]]/
0006 putself
0007 opt_send_without_block <callinfo!mid:str, argc:0, FCALL|VCALL|ARGS_SIMPLE>, <callcache>
0010 opt_send_without_block <callinfo!mid:===, argc:1, ARGS_SIMPLE>, <callcache>
0013 opt_not          <callinfo!mid:!, argc:0, ARGS_SIMPLE>, <callcache>
0016 trace            512
0018 leave
=> nil
irb(main):164:0>
```